test/types/mocha_decorations.d.ts

```diff
@@ -19,12 +19,27 @@
 
 import { Suite } from 'mocha';
 
+type Tags =
+  | 'ciGroup1'
+  | 'ciGroup2'
+  | 'ciGroup3'
+  | 'ciGroup4'
+  | 'ciGroup5'
+  | 'ciGroup6'
+  | 'ciGroup7'
+  | 'ciGroup8'
+  | 'ciGroup9'
+  | 'ciGroup10'
+  | 'ciGroup11'
+  | 'ciGroup12';
+
 // tslint:disable-next-line:no-namespace We need to use the namespace here to match the Mocha definition
 declare module 'mocha' {
   interface Suite {
     /**
      * Assign tags to the test suite to determine in which CI job it should be run.
      */
-    tags(tags: string[] | string): void;
+    tags<T extends Tags>(tags: T | T[]): void;
+    tags<T extends string>(tags: T | T[]): void;
```

Comment:
What's the difference between this and:

```ts
tags(tags: Tags | Tags[]): void;
tags(tags: string[]): void;
```

Reply:
Autocomplete 😀

Reply:
double declaration without generic get an error(fixed by combine types), but in this case, your IDE failed with autocomplete